src/urllib3/util/util.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional, Union
+
+
+def to_bytes(
+    x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None
+) -> bytes:
+    if isinstance(x, bytes):
+        return x
+    elif not isinstance(x, str):
+        raise TypeError(f"not expecting type {type(x).__name__}")
+    elif encoding or errors:
+        return x.encode(encoding or "utf-8", errors=errors or "strict")
+    return x.encode()
```

Comment:
You can remove one branch here:

```suggestion
    return x.encode(encoding or "utf-8", errors=errors or "strict")
```

Reply:
I think @sethmlarson suggested the branch in function to detect the case that we can call `x.encode()` because it's 4X faster. 
Take a look at this [comment](https://github.com/urllib3/urllib3/pull/2078#discussion_r527972377)